### PR TITLE
Name-with-icon component

### PR DESCRIPTION
**Pull request by buoyad**
Implements the [name with icon](https://zpl.io/VYJWXjJ) common adapter and adds stories for it. I also modified `ClickableBox` slightly (eek!) so it can take an array of styles and use `collapseStyles`. r? @keybase/react-hackers

### CODE REVIEW

**Comment by buoyad** [comment]
@keybase/react-hackers bump

File: shared/common-adapters/clickable-box.desktop.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ class ClickableBox extends React.Component<Props & {children: any}, {mouseDown:
         onMouseLeave={needMouseEnterLeaveHandlers(this.props) ? this._onMouseLeave : undefined}
         onMouseUp={this._onMouseUp}
         onClick={onClick}
-        style={{..._containerStyle, ...(onClick ? globalStyles.clickable : null), ...style}}
+        style={collapseStyles([_containerStyle, onClick ? globalStyles.clickable : {}, style || {}])}
```
Comment (chrisnojima) [inline shared/common-adapters/clickable-box.desktop.js]: you can keep null vs passing {} , this is desktop so you can spread null

Comment (chrisnojima) [reply shared/common-adapters/clickable-box.desktop.js]: same for the style || {} thing, just use style

File: shared/common-adapters/name-with-icon.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import {type IconType} from './icon.constants'
+import Text, {type TextType} from './text'
+import {Usernames} from './usernames'
+import {collapseStyles, globalStyles, isMobile, styleSheetCreate} from '../styles'
+
+type Size = 'small' | 'default' | 'large'
+
+// Exposed style props for the top-level container and box around metadata arbitrarily
+type Props = {
+  following?: boolean,
```
Comment (chrisnojima) [inline shared/common-adapters/name-with-icon.js]: lets not plumb through isYou/following/etc. Lets just use ConnectedUsername always

**Comment by buoyad** [comment]
I'm going to merge the implementation branch into this before continuing

File: shared/common-adapters/clickable-box.desktop.js (hunk 1 above)
Comment (buoyad) [reply shared/common-adapters/clickable-box.desktop.js]: On master `collapseStyles` is declared with `Array<Object>`, so I was getting errors. I'll switch it over once your changes are in

**Comment by buoyad** [comment]
ok @chrisnojima, added the connected avatar and got rid of those props. PTAL

**Comment by buoyad** [comment]
@keybase/react-hackers bump

**Pull request merged**